python/mxnet/ndarray.py

```diff
@@ -1054,6 +1054,45 @@ def array(source_array, ctx=None, dtype=None):
     return arr
 
 
+def moveaxis(tensor, source, destination):
+    """Moves the `source` axis into the `destination` position
+    while leaving the other axes in their original order
+
+    Parameters
+    ----------
+    tensor : mx.nd.array
+        The array which axes should be reordered
+    source : int or sequence of int
+        Original positions of the axes to move. These must be unique.
+    destination : int or sequence of int
+        Destination positions for each of the original axes. These must also be unique.
+
+    Returns
+    -------
+    result : mx.nd.array
+        Array with moved axes.
+
+    Examples
+    --------
+    >>> X = mx.nd.array([[1, 2, 3],
+                         [4, 5, 6]])
+    >>> mx.nd.moveaxis(X, 0, 1).shape
+    (3, 2)
+    """
+    axes = list(range(tensor.ndim))
+    try:
+        axes.pop(source)
+    except IndexError:
+        raise ValueError('Source should verify 0 <= source < tensor.ndim'
+                         'Got %d' % source)
+    try:
+        axes.insert(destination, source)
```

Comment:
I wrote the function to take `ints` as parameters and not lists so as to not overload the function. I can add support for lists of indices as well if you prefer.